### ISSUE TITLE
Add `Action` fields for plugin and samples menus

### ISSUE THREAD

**Issue by lucyleeow** [napari/napari#6224]
Plugin contributions that currently add items to the menubar:

* plugin widgets to the plugin menu
* samples to the file menu
* <FUTURE> contributable menu items (#6174, [NAP 6](https://napari.org/dev/naps/6-contributable-menus.html))

With the current implementation, contributable menu items have a lot more functionality by exposing these [`Action`](https://app-model.readthedocs.io/en/latest/types/) fields, that plugin and samples menus DO NOT:

* category
* tooltip
* icon
* enablement
* keybindings (not fully functional yet)

Implemented here:

https://github.com/napari/napari/blob/aee692d02eaa0169a9a99a25714752c19eb170c1/napari/plugins/_npe2.py#L545-L554

These are currently not implemented for samples in the file menu (#4865) and as such, I think will be out of scope for the plugin menu PR (#4991) as well. The `sample` and `widget` manifest contributions lack the above fields, so implementing these would involve obtaining both the `command` and `sample`/`widget` manifest contributions when building the sample/widget menus (currently we just iterate over e.g., `mf.contributions.sample_data`) - this makes things messier but very doable.

This issue is to highlight this and if we do want to add this, remind us to do it. This probably does not need to be done until after NAP-6 as some items (e.g., `enablement`) are not documented yet (but will be as part of NAP 6), even though technically it is implemented.

cc @DragaDoncila
